pkg/kubelet/cm/dra/manager.go

```diff
@@ -175,29 +171,42 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 				return fmt.Errorf("NodePrepareResources failed for claim %s/%s: %s", reqClaim.Namespace, reqClaim.Name, result.Error)
 			}
 
-			claimInfo := claimInfos[types.UID(claimUID)]
-
-			// Add the CDI Devices returned by NodePrepareResources to
-			// the claimInfo object.
-			err = claimInfo.addCDIDevices(pluginName, result.GetCDIDevices())
-			if err != nil {
-				return fmt.Errorf("failed to add CDIDevices to claimInfo %+v: %+v", claimInfo, err)
+			resourceClaim := resourceClaims[types.UID(claimUID)]
+			if !m.cache.contains(resourceClaim.Name, pod.Namespace) {
+				// Grab the allocation.resourceHandles. If there are no
+				// allocation.resourceHandles, create a single resourceHandle with no
+				// content. This will trigger processing of this claim by a single
+				// kubelet plugin whose name matches resourceClaim.Status.DriverName.
+				resourceHandles := resourceClaim.Status.Allocation.ResourceHandles
+				if len(resourceHandles) == 0 {
+					resourceHandles = make([]resourceapi.ResourceHandle, 1)
+				}
+				claimInfo := newClaimInfo(
+					resourceClaim.Status.DriverName,
+					resourceClaim.Spec.ResourceClassName,
+					resourceClaim.UID,
+					resourceClaim.Name,
+					resourceClaim.Namespace,
+					sets.New(string(pod.UID)),
+					resourceHandles,
+					false,
+				)
+				if err = m.cache.add(&claimInfo); err != nil {
+					return fmt.Errorf("failed to cache claimInfo %+v: %w", claimInfo, err)
+				}
 			}
```

Comment:
Looking again -- this won't work as is. It is racy with the addPodReference() code above.

Consider the two code blocks (which can run concurrently):
```
if cache.contains(...) {
    cache.addPodReference(...)
}
```
and
```
if cache.contains(...) {
    cache.add(...)
}
```

The following sequence is possible:
```
// pod 0
if cache.contains(claim0) // false
if cache.isPrepared(claim0) // false
if cache.contains(claim0) // false

// pod 1
if cache.contains(claim0) // false
if cache.isPrepared(claim0) // false
if cache.contains(claim0) // false

// pod 0
cache.add(claim0Info) // with reference to pod0

// pod 1
cache.add(claim0Info) // with reference to pod1
```
The entry in the cache now only has a reference to pod1 not pod0